modules/nf-core/ngscheckmate/ncm/tests/main.nf.test

```diff
@@ -0,0 +1,184 @@
+                            file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)
+                        ]
+                input[1] = [ [ id:'sarscov2' ],
+                         file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+                         ]
+                input[2] = false
+                """
+            }
+        }
+
+        run("BCFTOOLS_MPILEUP", alias: "BCFTOOLS_MPILEUP2") {
+            script "../../../bcftools/mpileup/main.nf"
+            process {
+                """
+                input[0] =  [
+                            [ id:'test2' ], // meta map
+                            file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
+                            file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)
+                        ]
+                input[1] = [ [ id:'sarscov2' ],
+                         file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+                         ]
+                input[2] = false
+                """
+            }
+        }
+
+    }
+
+
+    test("sarscov2 - bam") {
+        config "./bam.config"
+        when {
+            process {
+                """
+                input[0] = [ [ id: 'combined_bams' ],
+                         [file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam']               , checkIfExists: true ),
+                          file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai']           , checkIfExists: true ),
+                          file(params.test_data['sarscov2']['illumina']['test_paired_end_methylated_sorted_bam']    , checkIfExists: true ),
+                          file(params.test_data['sarscov2']['illumina']['test_paired_end_methylated_sorted_bam_bai'], checkIfExists: true )
+                          ]
+                     ]
+                input[1] = BEDTOOLS_MAKEWINDOWS.out.bed
+                input[2] = [ [ id:'sarscov2' ],
+                         file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+                         ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    process.out.corr_matrix,
+                    process.out.matched,
+                    process.out.all,
+                    process.out.versions
+                ).match() }
+            )
+        }
+
+    }
+
+    test("sarscov2 - vcf") {
+        config "./vcf.config"
+        when {
+            process {
+                """
+                input[0] = BCFTOOLS_MPILEUP1.out.vcf.combine(BCFTOOLS_MPILEUP2.out.vcf.map{it[1]}).map{meta, one, two -> [meta, [one, two]]}.map{meta, stuff -> [meta, stuff.flatten()]}
```

Comment:
This is horrific and could do with cleaning up...
I have two outputs that are both:
`[meta, vcf]`
and I want to make
`[new_meta, [vcf1, vcf2]]`